src/Qir/Runtime/public/QirContext.hpp

```diff
@@ -14,30 +14,33 @@ namespace Quantum
     struct IRuntimeDriver;
     struct AllocationsTracker;
 
-    QIR_SHARED_API void InitializeQirContext(IRuntimeDriver* driver, bool trackAllocatedObjects = false);
-    QIR_SHARED_API void ReleaseQirContext();
     struct QIR_SHARED_API QirExecutionContext
     {
-        IRuntimeDriver* driver = nullptr;
-        bool trackAllocatedObjects = false;
-        std::unique_ptr<AllocationsTracker> allocationsTracker;
+        static void Init(IRuntimeDriver* simulator, bool trackAllocatedObjects = false);
+        static void Deinit();
 
-        QirExecutionContext(IRuntimeDriver* sim, bool trackAllocatedObjects);
+        QirExecutionContext(IRuntimeDriver* simulator, bool trackAllocatedObjects);
         ~QirExecutionContext();
-    };
-    extern thread_local std::unique_ptr<QirExecutionContext> g_context;
-    extern QIR_SHARED_API std::unique_ptr<QirExecutionContext>& GlobalContext();
 
-    struct QIR_SHARED_API QirContextScope

```

Comment:
Removing this type is technically a breaking change, such that I think we should reconsider. Both our generated main.cpp and any clients that have started building against the runtime will need to change to adapt to this. I think we may need a more general discussion to identify which parts of our API are up for refactor vs which ones would be part for breaking change.

Reply:
I'm not removing this type, I'm encapsulating/moving it into `QirExecutionContext` and renaming to `Scoped`, i.e. now it is called `QirExecutionContext::Scoped` (I'm converting a separate class to a nested class).

The reason for encapsulating is that this class is built on top of `QirExecutionContext` class. If `QirExecutionContext` is moved or deleted, then it is reasonable for this class to be automatically moved or deleted together with the `QirExecutionContext`.

Reply:
I didn't know that our QIR code has customers.
Let's discuss that separately. Thank you for letting me know.